### ISSUE TITLE
Possible to connect to remote server?

### ISSUE THREAD

**Issue by tmeasday** [arunoda/meteor-ddp-analyzer#4]
Would it work to not force `localhost` as the upstream ddp endpoint?

**Comment by arunoda** [comment]
What do you trying to do? connect to a raw DDP connection or connect to a running app and access it locally? 

Both are possible.
- Currently, we can't use WebSocket with meteor apps, due to the fact DDP analyzer uses SockJS middleware.
- But when we need to connect to a raw DDP endpoint, we don't need to use the SockJS middleware (not implemented yet)
- If it is a running app, we need to parse initial HTML and add deafult ddp url to the runtime config. (possible, bit hard)

I think I can find sometime on next weekend on working on these features. Patches are welcome :)

**Comment by tmeasday** [comment]
I wanted to connect to a running app and watch the DDP traffic it's sending. I guess I'd need some easy way to send DDP messages also.

One option would be a bookmarklet that somehow injects code into the running app and re-routes the DDP socket to the proxy, given that the proxy is configured to connect to the server.

Anyway, I was just wondering if it was easy. If it's not, you can leave it on the wishlist :)

**Comment by arunoda** [comment]
Your suggestion is also good. But re-writing default ddp url in the index.html is not that hard too. I think, I can work on this in next weekend.

**Comment by tmeasday** [comment]
How would that work though? I mean supposing I just wanted to read the DDP traffic off a random site that I was connected to in my browser..

**Comment by arunoda** [comment]
instead of accessing the meteorapp.com, you are accessing it with localhost:3030. I parse the HTML and add `DDP_DEFAULT_CONNECTION_URL` with the ddp proxy URL.

Then you can access the site with `localhost:3030` and ddp logs with get printed.

**Comment by tmeasday** [comment]
Ah, yeah. Good idea :)

**Comment by tmeasday** [comment]
Although it may potentially break a bunch of stuff due to 3rd party shenanigans (3rd party fonts / XHR / etc).

**Comment by arunoda** [comment]
Yep. I thought so.
- XHR won't be an issue and most services uses zero image pixels
- fonts are normally configured to used with localhost (on typekit), so it will work too.
- This is identical as when the developer is working on the app locally

Some third party APIs might break. Let's give this a try and see.

**Comment by tmeasday** [comment]
Yeah definitely! I think the bookmarklet idea leads in the direction of a chrome extension. But that's a whole other story..

**Comment by arunoda** [comment]
Yeah. Chrome Extension is also a good idea. Then it can show the client side subscription stats and so on easily.

But it's a whole new project. I heard sometimes back, [avital](https://groups.google.com/forum/#!topic/meteor-talk/4C8ZEIZSFB8) talked about an chrome extension for blaze stats. Never heard about it after that.

This might go with that.
